File: app/src/main/java/com/kfang/fencemelibrary/Fencer.java
package com.kfang.fencemelibrary;

/**
 * Class to contain fencers
 */

class Fencer {
    private int points;
    private String name;
    private int redCards;
    private int yellowCards;
    private boolean hasPriority;

    // is decrement card functionality really needed?

    Fencer(String name) {
        this.points = 0;
        this.name = name;
        this.redCards = 0;
        this.yellowCards = 0;
    }

    void assignPriority() {
        this.hasPriority = true;
    }

    boolean hasPriority() {
        return this.hasPriority;
    }

    int getPoints() {
        return points;
    }

    public void setPoints(int points) {
        this.points = points;
    }

    void incrementNumPoints() {
        this.points++;
    }

    void decrementNumPoints() {
        this.points--;
    }

    public String getName() {
        return name;
    }

    void setName(String name) {
        this.name = name;
    }

    int getRedCards() {
        return redCards;
    }

    void setRedCards(int cards) {
        this.redCards = cards;
    }

    void incrementRedCards() {
        this.redCards++;
    }

    /* void decrementRedCards() {
        this.redCards--;
    } */

    int getYellowCards() {
        return yellowCards;
    }

    void setYellowCards(int cards) {
        this.yellowCards = cards;
    }

    void incrementYellowCards() {
        this.yellowCards++;
    }

    /*
    void decrementYellowCards() {
        this.yellowCards--;
    }
    */

    void resetCards() {
        this.redCards = 0;
        this.yellowCards = 0;
    }
}

File: app/src/main/java/com/kfang/fencemelibrary/model/Fencer.kt
package com.kfang.fencemelibrary.model

import android.databinding.BaseObservable
import android.databinding.Bindable

import com.kfang.fencemelibrary.BR

import java.io.Serializable


/**
 * Class to contain fencers
 */

class Fencer
// is decrement card functionality really needed?

(var defaultName: String) : BaseObservable(), Serializable {
    internal var points: Int = 0
    var redCards: Int = 0
    var yellowCards: Int = 0
    private var hasPriority: Boolean = false

    init {
        this.points = 0
        this.redCards = 0
        this.yellowCards = 0
    }

    fun assignPriority() {
        this.hasPriority = true
    }

    fun hasPriority(): Boolean {
        return this.hasPriority
    }

    @Bindable
    fun getPoints(): Int {
        return points
    }

    fun setPoints(points: Int) {
        this.points = points
        notifyPropertyChanged(BR.points)
    }

    fun incrementNumPoints() {
        this.points++
        notifyPropertyChanged(BR.points)
    }

    fun decrementNumPoints() {
        this.points--
        notifyPropertyChanged(BR.points)
    }

    var name: String
        @Bindable
        get() = defaultName
        set(name) {
            this.defaultName = name
            notifyPropertyChanged(BR.name)
        }

    fun incrementRedCards() {
        this.redCards++
    }

    fun incrementYellowCards() {
        this.yellowCards++
    }

    fun resetCards() {
        this.redCards = 0
        this.yellowCards = 0
    }
}
